service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/DeviceTwinDevice.java

```diff
@@ -536,6 +542,10 @@ public String toString()
         {
             thisDevice.append("Version: " + this.getVersion() + "\n");
         }
+        if(this.modelId != null && !this.modelId.isEmpty())
+        {
```

Comment:
Why is this check needed? What would happen if a null or empty is placed on parser.

Reply:
This is a toString function. There's no reason to print the modelId if there is no modelId

Reply:
I see that this pattern is followed everywhere. Looking back I would have just printed the data as is i.e. along with null to represent what has actually come on the wire from service. However now I don't think we can make that change and should follow the suit.

Reply:
If we want to print out null/empty values, this check is not needed. I followed the pattern similar to module id and others as it doesn't seem to provide any value printing out null/empty